README.md

```diff
@@ -23,25 +23,41 @@ use {'tzachar/fuzzy.nvim', requires = {'romgrk/fzy-lua-native'}}
 
 # Api
 
-This plugin supports a module with a single method: `filter`.
-The `filter` method receives three parameters: a pattern to match, a list of
-strings and an extra parameter passed to the fuzzy library: for fzy, its a
-boolean indicating `is_case_sensitive`, for fzf its the `case_mode` (see
-[this](https://github.com/nvim-telescope/telescope-fzf-native.nvim#lua-interface)). 
+This plugin supports a module with a single method: `filter`. The `filter`
+method receives three parameters: a pattern to match, a list of strings or
+tables where the first element is a string and an extra parameier passed to the
+fuzzy library: for fzy, its a boolean indicating `is_case_sensitive`, for fzf
+its the `case_mode` (see
+[this](https://github.com/nvim-telescope/telescope-fzf-native.nvim#lua-interface)).
 
-The method returns a list of `{string, match_positions, score}`. For the
-meanings of match_positions and score see individual documentations of `fzf` and
-`fzy`.
+The method returns a list of `{string|table, match_positions, score, index}`.
+For the meanings of match_positions and score see individual documentations of
+`fzf` and `fzy`. `index` is the index of the matched string in the input list.
```

Comment:
not clear what 'input list' means here. Refer to the arguments of the function to make it clear.